Implement Version Policy

PiyushRaj927 opened the pull request:
- Introduce changelog
- Update workflow configurations
- Pin TimescaleDB version
- Modify MAKEFILE to manage Warpsql versioning, prerelease, and generate Docker image tags

PiyushRaj927 commented:
Fixes #119

singhalkarun commented:
- Automatically sets PRE_RELEASE based on the current tag at the HEADof the git tree.

Can you detail this more, add examples?

- If no tag is present, it's set as dev-pg15.

Instead of making it dev-pg15, how about we have something like last commit id because the image will get lost out on a new commit?

PiyushRaj927 commented:
> * Automatically sets PRE_RELEASE based on the current tag at the HEADof the git tree.
> 
> 
> Can you detail this more, add examples?

I have updated the description to be more clear and added an example

> * If no tag is present, it's set as dev-pg15.
> 
> 
> Instead of making it dev-pg15, how about we have something like last commit id because the image will get lost out on a new commit?

sure, Makefile are updated to tag the images as per the format `dev-<HEAD SHA>-pg<version>` 

> * Documentations on Workflows
> 
> 
> Also, can we create a doc which explains how the workflow works in detail so that a someone exploring repository for first time also have a clear understanding how images get built.

I don't believe it's necessary to detail the specifics of the workflow since they're quite straightforward, and the image building and publishing logic is abstracted in Makefiles. Can you point out what parts of our current workflow setup that might pose challenges or appear opaque to newcomers? I'll try to make them clearer in the document.
